gef-ui/api-tests/facility-end-date.post.api-test.js

```diff
@@ -0,0 +1,199 @@
+const { MAKER } = require('../server/constants/roles');
+const { withRoleValidationApiTests } = require('./common-tests/role-validation-api-tests');
+const app = require('../server/createApp');
+const { post } = require('./create-api').createApi(app);
+const api = require('../server/services/api');
+const storage = require('./test-helpers/storage/storage');
+
+api.getFacility = jest.fn();
+api.updateFacility = jest.fn();
+api.updateApplication = jest.fn();
+
+const dealId = '123';
+const facilityId = '111';
+
+describe('facility-end-date routes', () => {
+  beforeEach(async () => {
+    resetAllWhenMocks();
+    await storage.flush();
+    jest.clearAllMocks();
+  });
+
+  afterAll(async () => {
+    resetAllWhenMocks();
+    jest.resetAllMocks();
+    await storage.flush();
+  });
+
+  describe('POST /application-details/:dealId/facilities/:facilityId/facility-end-date', () => {
+    beforeEach(() => {
+      when(api.getFacility)
+        .calledWith({ facilityId, userToken: expect.anything() })
+        .mockResolvedValueOnce({
+          details: {
+            isUsingFacilityEndDate: null,
+            coverStartDate: '2024-07-15T00:00:00.000Z',
+          },
+        });
+    });
+
+    describe('with saveAndReturn false', () => {
+      withRoleValidationApiTests({
+        makeRequestWithHeaders: (headers) =>
+          postFacilityEndDateWithHeaders({
+            body: { 'facility-end-date-year': '2024', 'facility-end-date-month': '8', 'facility-end-date-day': '12' },
+            headers,
+          }),
+        whitelistedRoles: [MAKER],
+        successCode: 302,
+        successHeaders: {
+          location: `/gef/application-details/${dealId}/facilities/${facilityId}/provided-facility`,
+        },
+      });
+    });
+
+    describe('with saveAndReturn true', () => {
+      withRoleValidationApiTests({
+        makeRequestWithHeaders: (headers) =>
+          postFacilityEndDateWithHeaders({
+            body: { 'facility-end-date-year': '2024', 'facility-end-date-month': '8', 'facility-end-date-day': '12' },
+            headers,
+            saveAndReturn: true,
+          }),
+        whitelistedRoles: [MAKER],
+        successCode: 302,
+        successHeaders: {
+          location: `/gef/application-details/${dealId}`,
+        },
+      });
+    });
+
+    describe('when the user is a maker', () => {
+      let sessionCookie;
+
+      beforeEach(async () => {
+        ({ sessionCookie } = await storage.saveUserSession([MAKER]));
+      });
+
+      it('returns 200 & does not update the database if the request body fails validation', async () => {
+        // Arrange
+        const body = { 'facility-end-date-year': '2023', 'facility-end-date-month': '8', 'facility-end-date-day': '12' };
+
+        // Act
+        const response = await postFacilityEndDateWithBodyAndSessionCookie({ body, sessionCookie });
+
+        // Assert
+        expect(response.status).toBe(200);
```

Comment:
sorry to be a pain could we reference status codes as their `HttpStatusCode` const from axios plz